librz/core/cmd/cmd_open.c

```
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+		char *t = rz_cmd_escape_arg(argv[i], RZ_CMD_ESCAPE_DOUBLE_QUOTED_ARG);
+		args[i - 1] = rz_str_newf("\"%s\"", t);
+		free(t);
+	}
+	char *args_str = rz_str_array_join((const char **)args, argc - 1, " ");
+	for (i = 0; i < argc - 1; i++) {
+		free(args[i]);
+	}
+	free(args);
+	rz_core_file_reopen_debug(core, args_str);
+	free(args_str);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_debug_file_handler(RzCore *core, int argc, const char **argv) {
+	const char *uri = argv[1];
+	ut64 addr = argc > 2 ? rz_num_math(core->num, argv[2]) : 0;
+	rz_core_file_reopen_remote_debug(core, uri, addr);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_debug_rzrun_handler(RzCore *core, int argc, const char **argv) {
+	char *file = rz_file_temp("rz-run");
+	char *s = strdup(argv[1]);
+	rz_config_set(core->config, "dbg.profile", file);
+	rz_str_replace_char(s, ',', '\n');
+	rz_file_dump(file, (const ut8 *)s, strlen(s), 0);
+	rz_file_dump(file, (const ut8 *)"\n", 1, 1);
+	free(file);
+	rz_core_file_reopen_debug(core, "");
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_malloc_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_file_reopen_in_malloc(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_nobin_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_file_reopen(core, NULL, 0, 0);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_nobin_write_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_file_reopen(core, NULL, RZ_PERM_RW, 0);
+	return RZ_CMD_STATUS_OK;
+}
+
+static RzCmdStatus reopen_nobin_headers(RzCore *core, int add_perms) {
+	RzIODesc *desc = rz_io_desc_get(core->io, core->file->fd);
+	if (!desc) {
+		RZ_LOG_ERROR("Could not find current file.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	int perms = core->io->desc->perm | add_perms;
+	char *fname = strdup(desc->name);
+	if (!fname) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_core_bin_load_structs(core, fname);
```

Check return values and handle errors in those functions please

---

Printed a warning, but I did not fail hard on this.